spec/requests/registrations_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'rails_helper'
+
+RSpec.describe 'Registrations', type: :request do
+  describe 'POST #create' do
+    subject(:request) { post '/signup.json', params: params }
```

Comment:
Can you use the route helper here? Also, when you want to make a POST request and expect a json response, you can define this on the headers rather than explicitly adding `.json` to the URL.